File: challenge-323/pokgopun/javascript/ch-1.js
//# https://theweeklychallenge.org/blog/perl-weekly-challenge-323/
/*#

Task 1: Increment Decrement

Submitted by: [47]Mohammad Sajid Anwar
     __________________________________________________________________

   You are given a list of operations.

   Write a script to return the final value after performing the given
   operations in order. The initial value is always 0.
Possible Operations:
++x or x++: increment by 1
--x or x--: decrement by 1

Example 1

Input: @operations = ("--x", "x++", "x++")
Output: 1

Operation "--x" =>  0 - 1 => -1
Operation "x++" => -1 + 1 =>  0
Operation "x++" =>  0 + 1 =>  1

Example 2

Input: @operations = ("x++", "++x", "x++")
Output: 3

Example 3

Input: @operations = ("x++", "++x", "--x", "x--")
Output: 0

Operation "x++" => 0 + 1 => 1
Operation "++x" => 1 + 1 => 2
Operation "--x" => 2 - 1 => 1
Operation "x--" => 1 - 1 => 0

Task 2: Tax Amount
#*/
//# solution by pokgopun@gmail.com

File: challenge-339/pokgopun/perl/ch-1.pl
### https://theweeklychallenge.org/blog/perl-weekly-challenge-339/
=begin

Task 1: Max Diff

Submitted by: [42]Mohammad Sajid Anwar
     __________________________________________________________________

   You are given an array of integers having four or more elements.

   Write a script to find two pairs of numbers from this list (four
   numbers total) so that the difference between their products is as
   large as possible.

   In the end return the max difference.

     With Two pairs (a, b) and (c, d), the product difference is (a * b)
     - (c * d).

Example 1

Input: @ints = (5, 9, 3, 4, 6)
Output: 42

Pair 1: (9, 6)
Pair 2: (3, 4)
Product Diff: (9 * 6) - (3 * 4) => 54 - 12 => 42

Example 2

Input: @ints = (1, -2, 3, -4)
Output: 8

Pair 1: (3, 1)
Pair 2: (-2, -4)

Example 3

Input: @ints = (-3, -1, -2, -4)
Output: 10

Pair 1: (-1, -2)
Pair 2: (-3, -4)

Example 4

Input: @ints = (10, 2, 0, 5, 1)
Output: 50

Pair 1: (10, 5)
Pair 2: (0, 1)

Example 5

Input: @ints = (7, 8, 9, 10, 10)
Output: 44

Pair 1: (10, 10)
Pair 2: (7, 8)

Task 2: Peak Point
=end
### solution by pokgopun@gmail.com
